### PR TITLE
document config mgmt proto

### PR DESCRIPTION
**Pull request by vjeffrey**
### :nut_and_bolt: Description: What code changed, and why?
Documenting the config mgmt proto for the api docs.

### :chains: Related Resources
https://github.com/chef/automate/issues/1801

### :+1: Definition of Done
config mgmt proto/responses/request message types are all documented.

### :athletic_shoe: How to Build and Test the Change
- check out this branch
- run make serve from the automate-chef-io directory
- visit http://localhost:1313/docs/api

### CODE REVIEW

File: api/external/cfgmgmt/cfgmgmt.proto [hunk 1]
```diff
@@ -25,41 +25,81 @@ import "components/automate-grpc/protoc-gen-policy/api/annotations.proto";
...
     option (chef.automate.api.policy).resource = "cfgmgmt:nodes:{node_id}:runs:{run_id}";
     option (chef.automate.api.policy).action = "read";
     option (chef.automate.api.iam.policy).resource = "infra:nodes:{node_id}";
     option (chef.automate.api.iam.policy).action = "infra:nodes:get";
   };
+
+  /*
+  GetSuggestions
+
+  Returns possible filter values given a valid `type` parameter and at least one character for the `text` parameter.
```
Comment (lancewf) [inline api/external/cfgmgmt/cfgmgmt.proto]: It is at least two characters for the text parameter or all values are returned. 
https://github.com/chef/automate/blob/82772a8c0a82da983bb9b329fb31716447d66609/components/config-mgmt-service/backend/elastic/suggestions.go#L121

Comment (vjeffrey) [reply api/external/cfgmgmt/cfgmgmt.proto]: 👍  thank you!

File: api/external/cfgmgmt/cfgmgmt.proto [hunk 2]
```diff
@@ -25,41 +25,81 @@ import "components/automate-grpc/protoc-gen-policy/api/annotations.proto";
...
     option (chef.automate.api.iam.policy).action = "infra:nodes:list";
   };
+
+  /*
+  GetRuns
+
+  Returns a list of run ids for the provided node id. 
+  Supports pagination.
+  Accepts a `start` parameter to denote start date for the list and a filter of type `status`.
+	*/
```
[the remark this answers is not in the document]
Comment (vjeffrey) [reply api/external/cfgmgmt/cfgmgmt.proto]: "run metadata" maybe? ill try that instead of run ids.

[the remark this answers is not in the document]
Comment (lancewf) [reply api/external/cfgmgmt/cfgmgmt.proto]: That sounds good.

File: api/external/cfgmgmt/cfgmgmt.proto [hunk 3]
```diff
@@ -18,48 +18,90 @@ import "api/external/common/version/version.proto";
...
 import "components/automate-grpc/protoc-gen-policy/iam/annotations.proto";
 
 service ConfigMgmt {
+  /*
+  GetNodes
+
+  Returns a list of infra nodes that have checked in to Automate. 
+  Adding a filter makes a list of all nodes that meet the filter criteria. 
+  Supports pagination, filtering, and sorting.
+	*/
```
Comment (kmacgugan) [inline api/external/cfgmgmt/cfgmgmt.proto]: Do we want to add information about how filters with the same field are "OR"ed together and filters across different fields are "AND"ed together?

Comment (vjeffrey) [reply api/external/cfgmgmt/cfgmgmt.proto]: ah ya, i can add a sentence or two about that! good idea

File: api/external/cfgmgmt/cfgmgmt.proto [hunk 4]
```diff
@@ -84,24 +138,39 @@ service ConfigMgmt {
...
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+			tags: "hidden";
+		};
   };
+
+  /*
+  GetPolicyCookbooks
+
+  Returns the policy information for a node.
+  The revision_id ??????
```
Comment (kmacgugan) [inline api/external/cfgmgmt/cfgmgmt.proto]: I would say something like: 
Returns Policy Names with a list of cookbook names and associated policy identifiers based on a policy revision ID. Policy revision ID's are sent with an infra run report.

File: api/external/common/query/parameters.proto [hunk 5]
```diff
@@ -4,22 +4,29 @@ package chef.automate.api.common.query;
...
+  // Field for which suggestions are being returned.
   string type = 1;
+  // Text to search on for the type value.
   string text = 2;
+  // Filters to be applied to the results.
   repeated string filter = 3;
 }
 
 enum SortOrder {
-  ASC = 0;   // By default we order ascending
```
Comment (kmacgugan) [inline api/external/common/query/parameters.proto]: Why remove the default information?

File: api/external/cfgmgmt/response/suggestions.proto [hunk 6]
```diff
@@ -4,6 +4,8 @@ package chef.automate.api.cfgmgmt.response;
 option go_package = "github.com/chef/automate/api/external/cfgmgmt/response";
 
 message Suggestion {
+  // Text to search on for the suggestions.
   string text = 1;
+  // Score of the result from ElasticSearch.
```
Comment (kmacgugan) [inline api/external/cfgmgmt/response/suggestions.proto]: Do we want to mention that a larger score is more heavily weighted?

File: api/external/cfgmgmt/cfgmgmt.proto [hunk 7]
```diff
@@ -18,48 +18,90 @@ import "api/external/common/version/version.proto";
...
     option (chef.automate.api.policy).action = "read";
     option (chef.automate.api.iam.policy).resource = "infra:nodes:{node_id}";
     option (chef.automate.api.iam.policy).action = "infra:nodes:list";
   };
+
+  /*
+  GetNodesCounts
+
+  Returns totals for failed, success, and missing infra nodes that have reported into Automate.
+	*/
```
Comment (kmacgugan) [inline api/external/cfgmgmt/cfgmgmt.proto]: Also returns overall total nodes

File: api/external/common/query/parameters.proto (hunk 5 above)
Comment (vjeffrey) [reply api/external/common/query/parameters.proto]: it's redundant: 
Screen Shot 2019-11-06 at 13 53 08

File: api/external/cfgmgmt/cfgmgmt.proto [hunk 8]
```diff
@@ -18,48 +18,90 @@ import "api/external/common/version/version.proto";
...
     option (chef.automate.api.policy).resource = "cfgmgmt:stats:run_counts";
     option (chef.automate.api.policy).action = "read";
     option (chef.automate.api.iam.policy).resource = "infra:nodes";
     option (chef.automate.api.iam.policy).action = "infra:nodes:list";
   };
+
+  /*
+  GetNodeRun
+
+  Returns the infra run report for the provided node id.
```
Comment (kmacgugan) [inline api/external/cfgmgmt/cfgmgmt.proto]: It is based on run id, as it is for a specific run.

File: api/external/cfgmgmt/response/suggestions.proto (hunk 6 above)
Comment (vjeffrey) [reply api/external/cfgmgmt/response/suggestions.proto]: hmm. i'm not sure it's super important information to say that (and also kinda obvious?) -- not sure how useful it is to the user -- @kagarmoe thoughts?

File: api/external/cfgmgmt/cfgmgmt.proto (hunk 4 above)
Comment (vjeffrey) [reply api/external/cfgmgmt/cfgmgmt.proto]: thank you!

**Pull request merged**